src/rdf_graph/rdf_extract.py

```diff
@@ -0,0 +1,970 @@
+                        help="The directory with Stanford JAR files.",
+                        default=None)
+
+    parser.add_argument('-sh', '--stanford_host',
+                        help="The Stanford host (e.g. 'http://localhost').",
+                        default=None)
+
+    parser.add_argument('-sp', '--stanford_port',
+                        type=int,
+                        help="The Stanford host port (e.g. 9000).",
+                        default=9000)
+
+    parser.add_argument('-sm', '--stanford_memory',
+                        help="The Stanford host memory (e.g. 'mx6g').",
+                        default='mx6g')
+
+    parser.add_argument('-st', '--stanford_timeout',
+                        type=int,
+                        help="The Stanford host timeout (e.g. 15000).",
+                        default=15000)
+
+    parser.add_argument('-rc', '--resolve_coreferences',
+                        action='store_true',
+                        help="Resolve the co-references.")
+
+    parser.add_argument('-pp', '--preprocess',
+                        default='abstract',
+                        choices=['abstract', 'document', 'none'],
+                        help="Preprocess the document or abstract.")
+
+    parser.add_argument('-pe', '--package_entities',
+                        action='store_true',
+                        help="Package the entities.")
+
+    args = parser.parse_args()
+
+    # initialize the spaCy object
+    try:
+        nlp = spacy.load('en_core_web_lg')
+    except OSError:
+        raise OSError('The model `en_core_web_lg` must be downloaded.')
+
+    with open(args.json_input) as fb:
+        docs = json.load(fb)
+
+    if args.preprocess == 'abstract':
+        docs = [preprocess_abstract(doc, nlp) for doc in docs]
+    elif args.preprocess == 'document':
+        docs = [preprocess_full_document(doc, nlp) for doc in docs]
+
+    if args.stanford_dir is not None and args.resolve_coreferences:
+        server = StanfordServer(args.stanford_dir,
+                                args.stanford_port,
+                                args.stanford_timeout,
+                                args.stanford_memory)
+        server.start()
+
+    if args.resolve_coreferences:
+        print("Resolving co-references...")
+        host = ('http://localhost'
+                if args.stanford_host is None or args.stanford_dir
+                else args.stanford_host)
+        resolver = StanfordCorefernceResolution(host,
+                                                args.stanford_port)
+        docs = resolver.resolve_all(docs)
+
+    print("Extracting triples...")
+    extractor = RDFExtractor(nlp,
+                             min_sub_char_len=args.min_sub_char_len,
+                             min_obj_char_len=args.min_obj_char_len,
+                             lemmatize=args.lemmatize,
+                             remove_numeric=args.remove_numeric)
+    extractor.extract_all(docs, raw=args.raw)
+
```

Comment:
So this is where we would potentially cluster our entities/relations based on their word embeddings. Check out the implementation here: https://github.com/GallupGovt/multivac/blob/d4484a517a9bed05265d6613391a557fc7b51291/src/rdf_graph/rdf_graph.py#L86 I'm wondering how hard it would be to drop that in here?